uplink/src/base/mod.rs

```diff
@@ -170,24 +180,58 @@ where
         self.last_sequence = current_sequence;
         self.last_timestamp = current_timestamp;
 
-        // Send buffer to serializer
-        if self.buffer.buffer.len() >= self.max_buffer_size {
-            let name = self.name.clone();
-            let topic = self.topic.clone();
-            let buffer = mem::replace(&mut self.buffer, Buffer::new(name, topic));
-            Ok(Some(buffer))
+        // if max_buffer_size is breached, flush
+        let buf = if self.buffer.buffer.len() >= self.max_buffer_size {
+            Some(self.flush_buffer())
         } else {
-            Ok(None)
+            None
+        };
+
+        Ok(buf)
+    }
+
+    // Returns buffer content, replacing with empty buffer in-place
+    fn flush_buffer(&mut self) -> Buffer<T> {
```

Comment:
This name along with `flush()` method is confusing. I'm defining `flush` as forwarding to serializer. Also we are not reusing this anywhere. Probably just leave `add` as is?

Reply:
Or use better name?